src/quo2/components/permission_tag.cljs

```diff
@@ -0,0 +1,125 @@
+   :border-radius    size
+   :justify-content  :center
+   :align-items      :center})
+
+(defn tag-resources [tokens size]
+  (let [tkns (take (if (> (count tokens) 3) 2 3) tokens)]
+    (for [{:keys [token-icon]} tkns]
+      ^{:key token-icon}
+      [react/view {:flex-direction :row}
+       [react/view (outer-resource-container size)
+        [react/image {:source token-icon
+                      :style  {:height        (case size 32 28 24 20)
+                               :width         (case size 32 28 24 20)
+                               :border-radius size}}]]])))
+
+(defn tag-extra-count [tokens size]
+  (let [taken-tkns (take (if (> (count tokens) 3) 2 3) tokens)
+        extra-counts (- (count tokens) (count taken-tkns))]
+    (when (> extra-counts 0)
+      [react/view {:flex-direction :row}
+       [react/view (outer-resource-container size)
+        [react/view (extra-count-styles size)
+         (if (< extra-counts 4)
+           [text/text {:size :label
+                       :style {:align-items     :center
+                               :color           (colors/theme-colors
+                                                 colors/neutral-50
+                                                 colors/neutral-40)}}
+            (str "+" extra-counts)]
+           [icons/icon :main-icons2/pending {:container-style {:align-items     :center
+                                                               :justify-content :center}
+                                             :color (colors/theme-colors
+                                                     colors/neutral-50
+                                                     colors/neutral-40)
+                                             :size  12}])]]])))
+
+(defn tag-or-clause [size]
+  [react/view {:align-items   :center}
+   [text/text {:weight         :medium
+               :style          {:size              (case size 32 :paragraph-2 24 :label)
+                                :color             (colors/theme-colors
+                                                    colors/neutral-50
+                                                    colors/neutral-40)
+                                :padding-left      4
+                                :text-transform :lowercase
+                                :padding-right     (case size 32 16 24 12)}}
+    (i18n/label :t/or)]])
+
+(defn tag []
+  (fn [{:keys [icon background-color icon-color token-groups size]
+        :or {size 24}}]
+    [react/view {:style (permission-tag-styles background-color size)}
+     [react/view {:padding-left    8
+                  :padding-right   (case size 32 16 24 12)}
+      [icons/icon icon
+       {:container-style {:align-items     :center
+                          :justify-content :center}
+        :resize-mode      :center
+        :size             (case size 32 20 24 16)
+        :color            icon-color}]]
+
+     (when (= (count token-groups) 1)
```

Comment:
Consider reducing this down to avoid code re-use. Something like:

```
 (let [left-group   ((first token-groups) :tokens)
             center-group (when  (= (count token-groups) 3) ((second token-groups ) :tokens))
             right-group (when  (>= (count token-groups) 2) ((last token-groups ) :tokens))
             ]
         [react/view {:flex-direction :row
                      :align-items    :center}
          (tag-resources left-group size)
          [tag-extra-count left-group size]
          (when (>= (count token-groups) 3)
            [tag-or-clause size]
          )
            (tag-resources center-group size)
            [tag-extra-count center-group size]
          
          (when (>= (count token-groups) 2)
            [tag-or-clause size]
          )
            (tag-resources right-group size)
            [tag-extra-count right-group size]
          
          ])]))
  ```
Albeit, perhaps there's a better alternative.